packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,192 @@
+| DX (Opinion)             | ✅ Similar to ``tag in HTML. Also we already have`isDefaultSelected`so might be intuitive to have`isSelected` as well                                                                                                                                       | Also intuitive because we support similar thing in other Inputs. My concerns was `value={['mango', 'apple']}` with multiselect. Passing array to `value` might seem a bit strange. Also this approach will only make sense if we overall [decide to make "ActionList" non-selectable items on consumer end](#are-actionlistitem--selectable-components) | This forces you to write selection logic outside of JSX which might not be as intuitive when you want to do something like fetching initial data from API and set selected values while rendering the ActionList. More info at [why not `useSelect`](#why-not-useselect) |
+
+#### Why not `useSelect`?
+
+Throughout initial discussions we mostly talked about creating this hook so noting down things here which made me change my opinion-
+
+- The hooks pattern for controlled inputs is not very common in ecosystem. Didn't find this being used anywhere apart from downshift-js (in their case, everything is hook). Most popular libraries follow something along the lines of `onChange` + `isSelected`.
+- Initial assumption was that there will be too much of state management on user's end with `onChange` + `isSelected` type of approach but now that I look at the [overall example](#onchange--isselected-), it doesn't seem too complex. It also covers most usecases I could think of like `reset`, reading currently selected options, etc.
+- Additional learning curve of `useSelect` API. Explaining `isSelected` is a lot more easier.
+- Since `useSelect` will return `selectOption` method, consumers have to hook this into their state management.
+
+```jsx
+const { values } = useFormik();
+const { selectOption } = useSelect();
+
+const doSomethingComplex = () => {
+  // ... bunch of code that updates `values`
+  selectOption(values.status);
+};
+
+// As opposed to in `onChange` it will just look like this -
+const { values } = useFormik();
+
+<ActionListItem isSelected={values.status === 'pending'} title="Pending" />;
+```
+
+### Referrences
+
+- Primer
+  - [ActionMenu](https://primer.style/react/ActionMenu#with-selection) (Closest to our approach. Uses `selected` and `onSelect`)
+    ```jsx
+    <ActionList.Item
+      key={index}
+      selected={index === selectedIndex}
+      //
+      onSelect={() => setSelectedIndex(index)}
```

Comment:
So their `<ActionList.Item />` is only "clickable" Its not used inside their Select. For them, `<SelectPanel />` is different component with completely different API - https://primer.style/react/SelectPanel

Here they are probably using "onSelect" as an alternative to "onClick" (probably because elements in dropdown can also be actioned by keyboard navigations and are not necessarily limited to clicks)